src/pages/api/signup.js

```diff
@@ -0,0 +1,51 @@
+import { PrismaClient } from "@prisma/client";
+import bcrypt from "bcrypt";
+
+const prisma = new PrismaClient();
+
+export default async function handler(req, res) {
+  if (req.method === "POST") {
+    const { username, email, password, role = "admin" } = req.body;
+
+    if (!username || !email || !password) {
+      return res
+        .status(400)
+        .json({ error: "Username, email, and password are required." });
+    }
+
+    try {
+      const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
take a look at this

https://www.quora.com/If-I-hash-passwords-before-storing-them-in-my-database-is-that-sufficient-to-prevent-them-being-recovered-by-anyone

also search about this " is hashing the password before storing it in the DB sufficient?"

Reply:
check this as well
https://markilott.medium.com/password-storage-basics-2aa9e1586f98